django/db/backends/sqlite3/schema.py

```diff
@@ -71,8 +93,10 @@ def alter_db_table(self, model, old_db_table, new_db_table, disable_constraints=
 
     def alter_field(self, model, old_field, new_field, strict=False):
         old_field_name = old_field.name
+        table_name = model._meta.db_table
+        _, old_column_name = old_field.get_attname_column()
         if (new_field.name != old_field_name and
-                any(r.field_name == old_field.name for r in model._meta.related_objects)):
+                self._is_referenced_by_fk_constraint(table_name, old_column_name, ignore_self=True)):
```

Comment:
We can ignore self-referential foreign key constraints because field alterations will require a table rebuild anyway which will cause the foreign key constraint to be reconstructed with the appropriate name.